server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1204,10 +1207,11 @@ public boolean attachIso(long isoId, long vmId, boolean forced) {
             throw new InvalidParameterValueException("Unable to find an ISO with id " + isoId);
         }
 
-        long dcId = vm.getDataCenterId();
-        VMTemplateZoneVO exists = _tmpltZoneDao.findByZoneTemplate(dcId, isoId);
-        if (null == exists) {
-            throw new InvalidParameterValueException("ISO is not available in the zone the VM is in.");
+        if (TemplateType.PERHOST.equals(iso.getTemplateType())) {
```

Comment:
if it was PERHOST we were not going to check, right? looks reversed here.

Reply:
@DaanHoogland thanks. Yeah made mistake